### PR TITLE
Feature/266

### LINKED ISSUE #266: As a guardian I would like to be able to edit a text to a pictogram in a weekplan so that other guardians know what I mean by it
There have been added some text to activities so the guardians can communicate with each other and the citizens about the activities. The guardians want to be able to change the text shown under the activities as shown in the image below. 

Right now it takes the name of the pictogram and uses as the standard text. This should still be the same and when the button hent standard is pressed this text should show up in the text field.

The button gem til borger should save the text in the field so the text for the activity changes to whats written in the field. It should also save the text so that all activities in the future that uses this pictogram for this citizen automatically gets this text instead of the standard text. Be aware that this saved text is specific for the citizen and only applies to activities that uses the pictogram in the future, not current activity that uses the pictogram.   
image

### PR DESCRIPTION
**Pull request by LJensen0403**
closes #266

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @Emil7370. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Comment by Determinatoro** [comment]
Please use the following checklist, @ProgDaniel. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

File: lib/screens/show_activity_screen.dart [hunk 1]
```diff
@@ -84,13 +87,15 @@ class ShowActivityScreen extends StatelessWidget {
         children: buildScreen(context, mode),
       );
     }
-    final bool keyboardVisible = MediaQuery.of(context).viewInsets.bottom > 0;
+    //final bool keyboardVisible = MediaQuery.of(context).viewInsets.bottom > 0;
```
Comment (ProgDaniel) [inline lib/screens/show_activity_screen.dart]: Remember to remove outcommented lines before commiting, if time allows

Comment (Rosmarium) [reply lib/screens/show_activity_screen.dart]: Fixed :)

**Comment by ProgDaniel** [comment]
When using a smaller device (a smartphone; used emulator is Pixel 2, 1080x1920; 420dpi), and being on the activity screen in landscape mode, the lower button 'Hent Standard' becomes overflowed at the bottom (displays 15 pixels)
image

Unsure if this is intended, but if you
1. create a custom title for an activity
2. change it to a choiceboard
3. add a secondary activity
4. delete the first, initial activity,

the remaining second activity will have the custom name of the first activity.

**Comment by ProgDaniel** [comment]
There are also 31 tests that fail when running 'flutter test'. Can you verify if they are related to this feature?

**Comment by Rosmarium** [comment]
> There are also 31 tests that fail when running 'flutter test'. Can you verify if they are related to this feature?

CI accepts, we are having no such problems here.

**Comment by ProgDaniel** [comment]
> CI accepts, we are having no such problems here.

after pulling the latest commits, the tests now all pass (save for one that skips), disregard the previous comment 👍

**Comment by AdrianPlesner** [comment]
>When using a smaller device (a smartphone; used emulator is Pixel 2, 1080x1920; 420dpi), and being on the activity screen in landscape mode, the lower button 'Hent Standard' becomes overflowed at the bottom (displays 15 pixels)

PO melder at dette problem ikke har prioritet lige nu, da det er meningen at appen skal anvendes på tablet devices. Der vil dog blive lavet et andet issue til at fixe det.

**Review by ProgDaniel — DISMISSED** [review]

**Review by ProgDaniel — APPROVED** [review]

**Review by Emil7370 — APPROVED** [review]

**Pull request merged**